__tests__/articlesRoutes.tests.js

```diff
@@ -88,6 +88,104 @@ describe("GET /api/articles", () => {
       expect(unsortedArticles).toEqual(articles);
     });
   });
+  describe("Query parameters", () => {
+    describe("Happy path", () => {
+      it("should filter articles by topic when topic query is given", async () => {
+        const {
+          body: { articles },
+        } = await request(app).get("/api/articles?topic=mitch").expect(200);
+        expect(articles).toHaveLength(11);
+        articles.forEach((article) => expect(article.topic).toBe("mitch"));
+      });
+      it("should sort artciles by the given query parameter", async () => {
+        const {
+          body: { articles },
+        } = await request(app)
+          .get("/api/articles?sort_by=article_id")
+          .expect(200);
+        const unsortedArticles = [...articles];
+        expect(unsortedArticles).toEqual(
+          articles.sort((a, b) => a.article_id + b.article_id)
+        );
+      });
+      it("should sort by ascending or descending if given the query parameter", async () => {
+        const {
+          body: { articles },
+        } = await request(app).get("/api/articles?order=desc").expect(200);
+        const unsortedArticles = [...articles];
+        expect(unsortedArticles).toEqual(
+          articles.sort(
+            (a, b) => new Date(b.created_at) - new Date(a.created_at)
+          )
+        );
+
+        const {
+          body: { articles: articles2 },
+        } = await request(app).get("/api/articles?order=asc").expect(200);
+        const unsortedArticles2 = [...articles2];
+        expect(unsortedArticles2).toEqual(articles.reverse());
+      });
+      it("should do all the query params at once 🤪", async () => {
+        const {
+          body: { articles },
+        } = await request(app)
+          .get("/api/articles?topic=mitch&sort_by=article_id&order=asc")
+          .expect(200);
+        expect(
+          articles.every((article, i, arr) => {
+            return i === 0 || article.article_id > arr[i - 1].article_id;
+          })
+        ).toBe(true);
+        articles.forEach((article) => expect(article.topic).toBe("mitch"));
+      });
+    });
+    describe("Sad path", () => {
+      it("should return an empty array when given a topic with no entries", async () => {
```

Comment:
Should we be expecting a 200 in this case?
I would say we have three possible scenarios for a topic value to consider:
- Topic exists and there are articles available, get those results.
- Topic exists but there are no articles available, get empty results.
- Topic does not exist
Can we separate these into tests and think about which is an error and which isn't and what status codes are appropriate